### PR TITLE
Add tests for Spain'17

### PR DESCRIPTION
**Pull request by ecarreras**
Add tests for Spain holidays in 2017

- [x] Tests with a significant number of years to be tested for your calendar.
- [ ] Changelog amended with a mention describing your changes.

### CODE REVIEW

**Comment by brunobord** [comment]
hem... why did you do that?

**Comment by Natim** [comment]
To make sure 2017 is not broken :?:

**Comment by ecarreras** [comment]
And why the others countries have the test for different years? If you
don't want you can close the PR

**Comment by Natim** [comment]
> And why the others countries have the test for different years?

For test coverage, when you have conditional holidays for instance.

Usually adding more tests if you don't fix anything doesn't make sense because it means more code to maintain, however for this specific library I would probably be in favor of adding test for each new year so that we make sure that we don't break previous working year and that the year to come is going to be working.

File: workalendar/tests/test_europe.py [hunk 1]
```diff
@@ -943,6 +943,18 @@ def test_year_2016(self):
         self.assertIn(date(2016, 12, 6), holidays)
         self.assertIn(date(2016, 12, 8), holidays)
 
+    def test_year_2017(self):
+        holidays = self.cal.holidays_set(2017)
+        self.assertIn(date(2017, 1, 1), holidays)
```
Comment (Natim) [inline workalendar/tests/test_europe.py]: Can we turn it into what @ewjohachim suggested in https://github.com/novafloss/workalendar/issues/170#issuecomment-270161567?

```
self.assertEqual(holidays_set(), {
    date(2013, 12, 25),  # Christmas
    # ...
})
```

**Comment by brunobord** [comment]
> And why the others countries have the test for different years? If you don't want you can close the PR

It followed this workflow:

* a calendar for country "X" is created in the year 2015,
* tests include holidays for year 2015,
* there are some specific dates to compute for a specific year (e.g. in the US, the presidential election day is a non-working day, it happens only every 4 years),
* tests are included for the specific year

When the contribution is merged, the code and the tests are validating the calendar *at the day it's implemented*. That's the main reason why some calendars have tests in several years.

What can happen next: if at some point the calendar is invalid for the year YYYY (changes in some computation day, new holiday, a holiday is removed...):

* we open a PR with the test case of the YYYY with a different holiday set (to mark the changes),
* these tests **should fail**
* the next commit would be the implementation of the changes
  * a - it should pass for year YYYY
  * b - it shouldn't break past years

I think we should keep the test suite as minimal as possible, to avoid having to compute every year since the beginning of calendars ; that'd be overkill.

On the other hand, if at some point, Spain calendar specifics change for some reason, then we'll add the specific cases to the tests and check that past, present and future years are handled correctly.

**Comment by xcambar** [comment]
> I would probably be in favor of adding test for each new year

I'd love to read your tests for a function testing if an integer is odd.

**Comment by Natim** [comment]
@xcambar :D

**Pull request closed without merging**